src/LiveComponent/src/EventListener/QueryStringInitializeSubscriber.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\UX\LiveComponent\EventListener;
+
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\UX\LiveComponent\Metadata\LiveComponentMetadata;
+use Symfony\UX\LiveComponent\Metadata\LiveComponentMetadataFactory;
+use Symfony\UX\LiveComponent\Util\QueryStringPropsExtractor;
+use Symfony\UX\TwigComponent\Event\PreMountEvent;
+
+/**
+ * @author Nicolas Rigaud <squrious@protonmail.com>
+ *
+ * @experimental
+ *
+ * @internal
+ */
+class QueryStringInitializeSubscriber implements EventSubscriberInterface
+{
+    public function __construct(
+        private readonly RequestStack $requestStack,
+        private readonly LiveComponentMetadataFactory $metadataFactory,
+        private readonly QueryStringPropsExtractor $queryStringPropsExtractor,
+    ) {
+    }
+
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            PreMountEvent::class => 'onPreMount',
```

Comment:
Can you add some tests to see how the different listener interfere (or not) with this listener ? 

I think IntreceptChild / Defer / DataModel  but maybe more, maybe less

And i think we should add priorities to ensure a determined order.

Reply:
I don't think it would be useful to test with InterceptChild and Defer, as they don't subscribe to the PreMount event. 

For DataModel, I'll have to setup some advanced use cases with parent and child components in order to test it efficiently. 

My first thought is the query string initialization should come before the DataModel listener. The latter purpose is to set the child prop value from the parent bound prop, and I think replacing this value with the one extracted from the query string will lead the child component in an inconsistent state regarding its parent. Wdyt?